### DISCUSSION TITLE
UGS Simulation Mode

### DISCUSSION THREAD

**Discussion by bobgibson2908** [winder/Universal-G-Code-Sender#2539] [Help]
I have a gCode that crashes my CNC at the same point each time I run it. Is it possible to run UGS in simulation mode to help me identify what is causing the problem?

**Comment by breiler** [comment]
No, there is no simulation in UGS. Is it physically crashing the machine or is it triggering some sort of fault on the machine?

If you post your gcode I can have a quick look.

**Reply by bobgibson2908** [reply]
It looks to me like rouge gcode to me. Tried to attach the .nc file but its not accepted. How can I send it to you

**Reply by breiler** [reply]
I think that you can ZIP it and then drag'n'drop it here.

**Reply by bobgibson2908** [reply]
[CELTIC KNOT 2 Finish.nc.zip](https://github.com/user-attachments/files/15747340/CELTIC.KNOT.2.Finish.nc.zip)
Here it is I've been away for a few days. I think the crash is around line 2140 its pretty basic I'm still very much a learner

**Reply by breiler** [reply]
I am assuming that you've done a rough cut and are not doing a 16mm cut here?
![Screenshot from 2024-06-08 14-01-34](https://github.com/winder/Universal-G-Code-Sender/assets/8962024/656bfcd2-7e95-4ea1-aef1-59ac27cb2c67)

**Reply by bobgibson2908** [reply]
yes rough cut already done..Leaving 1mm.  At the trouble spot the X & Y axis seen to stop and the Z axis travels down and bottoms out on the limit switch

**Reply by breiler** [reply]
I can't find any problems with the gcode, the lowest point it should travel is -16mm. I have no idea what's going on.

**Reply by bobgibson2908** [reply]
![IMG_7389](https://github.com/winder/Universal-G-Code-Sender/assets/43304449/56ccb9fa-7dec-46f2-b7dc-58f1ab5f491d)
Here is a photo of the actual piece. the block is 85mmX85mm X 19mm
I use Carbide Create to design the job. The rough cut went down 3mm and was to leave 1mm for the finish cut to deal with.
At no point was I expecting any cuts of 16mm . In the photo you can see the hole that I suspect is the plunge start of a deep cut that I'm not expecting. What is the gCode line number where it goes down 16mm? Ive set the job up twice in Carbide Create and it looks fine in the simulation. If you dont have any other suggestion I will just bin the project. I really appreciate the help you have provided.
Cheers Bob

**Reply by breiler** [reply]
There was no image attached, github does not allow attachments if you answer by email.

Here is the first gcode on line 2096 that will plunge down -16mm:
![image](https://github.com/winder/Universal-G-Code-Sender/assets/8962024/172e8496-fbc5-4235-b7ac-1f5108cff74e)

**Reply by breiler** [reply]
Yes, that hole corresponds well with what the gcode says. You need to keep tweaking the settings in Carbide Create.

**Comment by bobgibson2908** [comment]
I forgot to attach it.

are it is

￼
